flash.sh

```diff
@@ -0,0 +1,298 @@
+ADB=${ADB:-adb}
+VARIANT=${VARIANT:-eng}
+FULLFLASH=false
+
+if [ ! -f "`which \"$ADB\"`" ]; then
+	ADB=out/host/`uname -s | tr "[[:upper:]]" "[[:lower:]]"`-x86/bin/adb
+fi
+if [ ! -f "`which \"$FASTBOOT\"`" ]; then
+	FASTBOOT=out/host/`uname -s | tr "[[:upper:]]" "[[:lower:]]"`-x86/bin/fastboot
+fi
+
+run_adb()
+{
+	$ADB $ADB_FLAGS $@
+}
+
+update_time()
+{
+	if [ `uname` = Darwin ]; then
+		OFFSET=`date +%z`
+		OFFSET=${OFFSET:0:3}
+		TIMEZONE=`date +%Z$OFFSET|tr +- -+`
+	else
+		TIMEZONE=`date +%Z%:::z|tr +- -+`
+	fi
+	echo Attempting to set the time on the device
+	run_adb wait-for-device &&
+	run_adb shell "toolbox date $(date +%s) ||
+	               toolbox date -s $(date +%Y%m%d.%H%M%S)" &&
+	run_adb shell setprop persist.sys.timezone $TIMEZONE
+}
+
+sd_partition()
+{
+  DRIVE=$2
+
+  if [ -z "${DRIVE}" ]; then
+    echo You must specify a SD card drive
+  fi
+
+  umount $DRIVE
+
+  sudo dd if=/dev/zero of=$DRIVE bs=1024
+
+  sed -e 's/\t\([\+0-9a-zA-Z]*\)[ \t].*/\1/' << EOF | fdisk ${DRIVE}
+    o
+    n
+    p
+    1
+
+    +48M
+    n
+    p
+    2
+
+    +537M
+    n
+    p
+    3
+
+    +128M
+    n
+    p
+    4
+
+
+    a
+    1
+    p
+    w
+    q
+  EOF
+
+
+  case `uname` in
+  "Darwin")
+    e2fsck -f ${DRIVE}s4
+    ;;
+  "Linux")
+    e2fsck -f ${DRIVE}p4
```

Comment:
nit: since the only difference between osx and linux is s4/p4 we can just set this:
case `uname` in
  "Darwin")
    slice="s4"
    ;;
[...]
  esac
e2fsck -f ${DRIVE}${slice}